tests/button.spec.ts

```diff
@@ -0,0 +1,95 @@
+})
+
+test('Renders a button in the style of the primary variant if variant is specified as primary', async ({ page }) => {
+  const primaryButton = page.getByTestId('primary')
+  await expect(primaryButton.locator('button')).toBeVisible()
+  await expect(primaryButton.locator('button')).toHaveClass(/bg-white/)
+  await expect(primaryButton.locator('button')).toHaveClass(/border-gray-9/)
+
+})
+
+test('Renders a button in the style of inverse primary if the variant is specified as inverse primary', async ({ page }) => {
+  const inversePrimaryButton = page.getByTestId('inverse-primary')
+  await expect(inversePrimaryButton.locator('button')).toBeVisible()
+  await expect(inversePrimaryButton.locator('button')).toHaveClass(/bg-gray-9/)
+  await expect(inversePrimaryButton.locator('button')).toHaveClass(/border-gray-9/)
+  await expect(inversePrimaryButton.locator('button')).toHaveClass(/text-white/)
+})
+
+test('Renders a button in the style of success if the variant is specified as success', async ({ page }) => {
+  const successButton = page.getByTestId('success')
+  await expect(successButton.locator('button')).toBeVisible()
+  await expect(successButton.locator('button')).toHaveClass(/bg-success-fg/)
+  await expect(successButton.locator('button')).toHaveClass(/border-success-fg/)
+  await expect(successButton.locator('button')).toHaveClass(/text-white/)
+})
+
+test('Renders a button in the style of danger if the variant is specified as danger', async ({ page }) => {
+  const dangerButton = page.getByTestId('danger')
+  await expect(dangerButton.locator('button').first()).toBeVisible()
+  await expect(dangerButton.locator('button').first()).toHaveClass(/bg-danger-fg/)
+  await expect(dangerButton.locator('button').first()).toHaveClass(/border-danger-fg/)
+  await expect(dangerButton.locator('button').first()).toHaveClass(/text-white/)
+})
+
+test('Renders a button in the style of outline danger if the variant is specified as outline danger', async ({ page }) => {
+  const outlineDangerButton = page.getByTestId('outline-danger')
+  await expect(outlineDangerButton.locator('button')).toBeVisible()
+  await expect(outlineDangerButton.locator('button')).toHaveClass(/bg-danger-bg/)
+  await expect(outlineDangerButton.locator('button')).toHaveClass(/border-danger-bg/)
+  await expect(outlineDangerButton.locator('button')).toHaveClass(/text-danger-fg/)
+})
+
+test('Renders a button as disabled if the button has a disabled attribute of true', async ({ page }) => {
+  const disabledButton = page.getByRole('button', { name: 'Disabled Label' })
+  await expect(disabledButton).toBeVisible()
+  await expect(disabledButton).toBeDisabled()
+})
+
+test('Renders a tooltip over the button if a tooltip attribute is specified', async ({ page }) => {
+  const tooltipButton = page.getByRole('button', { name: 'Tooltip Label' })
+  const tooltipText = page.getByRole('tooltip', { name: 'This is the message for the tooltip'})
```

Comment:
nit: not necessary here - because we already filter by the text? May be a better idea to jut filter for a tooltip then check the text below?